ai_genomics/analysis/tag_evaluation/add_oa_tag_thresholds.py

```diff
@@ -0,0 +1,73 @@
+"""Script to add OA tag thresholds in labelled tag extraction validation dataset."""
+import ast
+import pandas as pd
+
+from ai_genomics.getters.data_getters import load_s3_data, save_to_s3
+from ai_genomics import bucket_name
+from ai_genomics.analysis.tag_evaluation.make_tag_validation_dataset import (
+    load_sample_tags,
+)
+
+LABELLED_TAG_EVALUATION_DATASET = load_s3_data(
+    bucket_name,
+    "inputs/ai_genomics_samples/labelled_ai_genomics_sample_validation_set.csv",
+)
+
+
+def get_oa_tags_above_threshold(
+    oa_samples: pd.DataFrame, threshold: int
+) -> pd.DataFrame:
+    """Generates column of OpenAlex tags above a threshold."""
+    oa_samples["scores"] = oa_samples["scores"].apply(
+        lambda x: [i for i, score in enumerate(x) if score > threshold]
+    )
+    oa_samples[f"oa_tags_above_{threshold}"] = [
+        [tags[i] for i in tag_indx]
+        for tag_indx, tags in zip(oa_samples["scores"], oa_samples["tags"])
+    ]
+
+    return oa_samples
+
+
+if __name__ == "__main__":
+
+    oa_samples = load_sample_tags()
+    labelled_dataset_oa_scores = pd.merge(
+        LABELLED_TAG_EVALUATION_DATASET, oa_samples, on="description"
+    )
+
+    for col in ("scores", "tags"):
+        labelled_dataset_oa_scores[col] = labelled_dataset_oa_scores[col].apply(
+            ast.literal_eval
+        )
+
+    labelled_dataset_oa_scores.scores = labelled_dataset_oa_scores.scores.apply(
+        lambda x: [float(i) for i in x]
+    )
```

Comment:
You can apply conversions like this on loading a csv with pandas. See [the `converters` parameter here](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html).

Reply:
cool - i've never used this parameter